src/components/Tutorial/TutorialSwap/constant.ts

```diff
@@ -0,0 +1,48 @@
+import { t } from '@lingui/macro'
+import { Step } from 'walktour'
+
+export interface StepCustom extends Step {
+  stopPropagationMouseDown?: boolean // stop Propagation onMouseDown event, ex: prevent Menufly setting auto close
+  center?: boolean // popup at center
+  hasPointer?: boolean
+  spotlightInteraction?: boolean
+  pcOnly?: boolean
+  popupStyle?: React.CSSProperties
+  requiredClickSelector?: string // click other element before enter this step
+  selectorHint?: string // this is element to check we clicked requiredClickSelector or not.
+  stepNumber?: number // for tracking, display
+  callbackEndStep?: () => void
+}
+
+export const TOTAL_STEP = 8
```

Comment:
Dynamic calculate it

Reply:
No.   It should be a fixed number. because we have 8 step but step 4, it's include 2 step.   so, I can not use array.lenght , ...

Reply:
I dont think so. It have to be some ways to dynamic calculate it. How about  max `stepNumber` of list steps?

Reply:
I think we should accept it.

`TOTAL_STEP =  getListSteps(true).reduce((a, b) => Math.max(a, b.stepNumber), 0)     // 8`

It will make same result.
but I don't prefer it because `getListSteps()` create some react element in it. we create that list and calculate only `stepNumber`, I think we may waste memory.